### PR TITLE
v5 - Add userPhoneNumber property to PayPalRequest

### PR DESCRIPTION
**Pull request by tdchow**
### Summary of changes

 - Add `userPhoneNumber` property to `PayPalRequest`
 - Update demo app to accept country code and national number fields
 - Update `PayPalCheckoutRequest` and `PayPalVaultRequest` to send phone number in the request body
 - Add `PayPalPhoneNumber` class that contains national number and country code
 
 
Screenshot_20240820_110936

I tested both checkout and vault flows. The phone number info is sent successfully.

Screenshot 2024-08-20 at 10 38 50 AM

Screenshot 2024-08-20 at 10 38 33 AM

### Checklist

 - [X] Added a changelog entry
 - [X] Relevant test coverage

### CODE REVIEW

**Comment by tdchow** [comment]
Closing PR until the release plan is set.

**Comment by tdchow** [comment]
I retested passing buyer phone number for both vault and checkout flows. The params are getting sent in the order create APIs.

```
  "phone_number": {
    "country_code": "312",
    "national_number": "1231234"
  }
```

**Review by jaxdesmarais — APPROVED** [review]

File: Demo/src/main/res/layout/fragment_paypal.xml [hunk 1]
```diff
@@ -21,6 +21,32 @@
...
+    <com.google.android.material.textfield.TextInputLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+
+        android:hint="@string/paypal_phone_country_code">
+
+        <com.google.android.material.textfield.TextInputEditText
+            android:id="@+id/buyer_phone_country_code_edit_text"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content" />
```
Comment (richherrera) [inline Demo/src/main/res/layout/fragment_paypal.xml:33]: What if we set the input type to `phone`?

File: Demo/src/main/res/layout/fragment_paypal.xml [hunk 2]
```diff
@@ -21,6 +21,32 @@
...
+    <com.google.android.material.textfield.TextInputLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginBottom="20dp"
+        android:hint="@string/paypal_phone_national_number">
+
+        <com.google.android.material.textfield.TextInputEditText
+            android:id="@+id/buyer_phone_national_number_edit_text"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content" />
```
Comment (richherrera) [inline Demo/src/main/res/layout/fragment_paypal.xml:46]: same here, what do you think?

**Review by richherrera — APPROVED** [review]

File: Demo/src/main/res/layout/fragment_paypal.xml (hunk 1 above)
Comment (tdchow) [reply Demo/src/main/res/layout/fragment_paypal.xml]: Great suggestion! I just pushed the change.

File: Demo/src/main/res/layout/fragment_paypal.xml (hunk 2 above)
Comment (tdchow) [reply Demo/src/main/res/layout/fragment_paypal.xml]: Updated!

**Pull request merged**